flycheck.el

```diff
@@ -7868,6 +7869,25 @@ See https://github.com/processing/processing/wiki/Command-Line"
   ;; This syntax checker needs a file name
   :predicate (lambda () (buffer-file-name)))
 
+(flycheck-define-checker protobuf-protoc
+  "A protobuf syntax checker using the protoc compiler.
+
+See URL `https://developers.google.com/protocol-buffers/'."
+  :command ("protoc" "--error_format" "gcc"
+            (eval (concat "--java_out=" (flycheck-temp-dir-system)))
+            (eval (concat "--proto_path=" (file-name-directory (buffer-file-name))))
+            source-inplace)
+  :error-patterns
+  ((info line-start (file-name) ":" line ":" column
+         ": note: " (message) line-end)
+   (error line-start (file-name) ":" line ":" column
+          ": " (message) line-end)
+   (error line-start
+          (message "In file included from") " " (file-name) ":" line ":"
+          column ":" line-end))
+  :modes protobuf-mode
+  :predicate (lambda () (buffer-file-name)))
```

Comment:
I think so, but I like the explicit lambda for the sake of syntactic consistency.